src/lib/libast/meson.build

```diff
@@ -195,6 +195,10 @@ if cc.has_header('mach-o/dyld.h')
     libast_c_args += [ '-D_hdr_mach_o_dyld']
 endif
 
+if not cc.has_function('iconv_open', prefix : '#include <iconv.h>')
```

Comment:
Move the `cc.find_library()` checks below ahead of this code and add `dependencies: iconv_dep` this test.